lms/services/assignment.py

```diff
@@ -20,6 +20,11 @@ class AssignmentService:
     def __init__(self, db: Session):
         self._db = db
 
+    def get_assignment_by_id(self, assignment_id) -> Assignment:
+        """Get an assignment by id."""
+
+        return self._db.query(Assignment).get(assignment_id)
```

Comment:
Once we know the assignment we want, we don't have to keep passing indirect identifiers about.